main.js

```diff
@@ -0,0 +1,135 @@
+const dropdown = document.querySelector("select");
+dropdown.addEventListener("change", sortOnChange);
+document.getElementById('taskInput').addEventListener("keydown",insert);
+let storage = {};
+
+if(localStorage.getItem("storage") === null){
+    localStorage.setItem("storage", JSON.stringify({items:[], idCount:0, sortBy:"date"}));
+    storage = JSON.parse(localStorage.getItem("storage"));
+}
+else{
+    storage = JSON.parse(localStorage.getItem("storage"));
+    dropdown.value = storage.sortBy;
+    let newItems = sort(storage.items, storage.sortBy);
+    for(let i = 0; i<newItems.length; i++){
+        itemToUl(newItems[i],0);
+    }
+}
+
+function insert(e){
+    if(document.getElementById('taskInput').value !== "" && e.keyCode === 13) {
+        let task = document.getElementById('taskInput').value;
+        document.getElementById('taskInput').value = "";
+        let today = new Date();
+        let newTask={"task":task, "date":today, "checkBox":false, "id":storage.idCount};
+        storage.idCount++;
+        storage.items.push(newTask);
+        localStorage.setItem("storage",JSON.stringify(storage));
+        let sortedItems = sort(storage.items,storage.sortBy );
+        for(let i=0;i<sortedItems.length;i++){
+            if (sortedItems[i].id === newTask.id){
+                itemToUl(newTask,i)
+            }
+        }
+    }
+}
+
+function sort(items, sortType){
+    if (sortType === "alphabetical order") {
+        items.sort(function(a, b) {
+            let taskA = a.task.toUpperCase(); // ignore upper and lowercase
+            let taskB = b.task.toUpperCase(); // ignore upper and lowercase
+            if (taskA < taskB) {
+                return -1;
+            }
+            if (taskA > taskB) {
+                return 1;
+            }
+            return 0;
+        });
+    }
+
+    else if (sort === "done"){
+        items.sort(function(a, b) {
+            let doneA = a.checkBox;
+            let doneB = b.checkBox;
+            if (doneA > doneB) {
+                return -1;
+            }
+            if (doneA < doneB) {
+                return 1;
+            }
+            return 0;
+        });
+    }
+    else{      //sorting by date, defult
+        items.sort(function(a, b) {
+            let dateA = new Date(Date.parse(a.date));
+            let dateB = new Date(Date.parse(b.date));
+            if (dateA < dateB) {
+                return -1;
+            }
+            if (dateA > dateB) {
+                return 1;
+            }
+            return 0;
+        });
+    }
+    return(items);
+}
+
+function changeCheckbox(id) {
```

Comment:
when checking checkbox and "sort by" is set to "done", should append the list element to the "done" elements list